cli/commands/create_benchmark_pipeline_command.py

```diff
@@ -0,0 +1,164 @@
+                f"mcs_benchmark_data.{template_type.value}_template_metadata"
+            )
+            TemplateDataclass = getattr(
+                template_module, f"{template_type.value.capitalize()}TemplateMetadata"
+            )
+
+            template_metadata = TemplateDataclass(benchmark_name=benchmark_name)
+
+            with open(
+                path_to_templates / template_metadata.template_name
+            ) as template_file:
+                template_str = template_file.read()
+
+            if template_type == TemplateType.METADATA:
+                temp = Template(template_str)
+                formatted_str = temp.substitute(**format_args)
+
+            else:
+                formatted_str = template_str.format(**format_args)
+
+            with open(
+                root_path / template_metadata.dest_file_path_from_root, "w"
+            ) as fp:
+                fp.write(formatted_str)
+
+            if template_type == TemplateType.METADATA:
+
+                with open(
+                    root_path
+                    / f"test_{str(template_metadata.dest_file_path_from_root)}",
+                    "w",
+                ) as fp:
+                    fp.write(formatted_str)
+
+    def __make_benchmark_directories(self, root_path: Path, benchmark_name: str):
+
+        root_path = Path(__file__).parent.parent.parent
+
+        for dataset_type in DatasetType:
+            os.makedirs(
+                root_path / "data" / benchmark_name / "datasets" / dataset_type.value
+            )
+            os.makedirs(
+                root_path
+                / "test_data"
+                / benchmark_name
+                / "datasets"
+                / dataset_type.value
+            )
+
+        self._logger.info(
+            f"""6 new directories have been made for the dev, train, and test datasets of the new benchmark in the following paths:\n
+        -./data/{benchmark_name}/datasets/{dataset_type.value}\n
+        -./test_data/{benchmark_name}/datasets/{dataset_type.value}\n
+            Please add the proper data files to each respective directory. Test_data files should have a smaller subset of data to facilitate expedited testing.\n
+            Additionally, edit the metadata.json file from ./data/benchmark_name by filling in the names of the benchmark authors.\n"""
+        )
+
+        # Make pipeline directory
+
+        path_to_pipeline = (
+            root_path / "mcs_benchmark_data" / "pipelines" / benchmark_name
+        )
+
+        os.makedirs(path_to_pipeline)
+
+        Path(path_to_pipeline / "__init__.py").touch()
+
+        self._logger.info(
+            f"""A new directory has been made for the pipeline files of the new benchmark at the following path:\n
+        -./mcs_benchmark_data/pipelines/{benchmark_name}\n
+        Please edit the {benchmark_name}_benchmark_* files according to the steps specified in the README.md"""
+        )
+
+        self._logger.info(

```

Comment:
`logging` has its own formatting mechanism with the old-style %s and variadic arguments. Standard design to avoid creating formatted log message strings for the common case where fine-grained logging is going to be discarded.

See the `logging` module for documentation.

Reply:
Admittedly not crucial to this case, but worth being familiar with.